### DISCUSSION TITLE
Front-end

### DISCUSSION THREAD

**Discussion by parthdacoder** [huggingface/chat-ui#724] [General]
Can you tell how can one just use the front-end for own development of chat-ui with a backend which is based on python?

**Comment by nsarrazin** [comment]
We support [a bunch of different backend options](https://github.com/huggingface/chat-ui#custom-models), but I think if you want to implement your own, the easiest way might be to [match the OpenAI spec](https://github.com/huggingface/chat-ui#openai-api-compatible-models). 

Let me know if that answers your question!

**Reply by parthdacoder** [reply]
Is it possible to just get the front-end? In order to get it which files need to be kept and which to be removed? 
I already tried to get it by removing the server folder, it is hosting but nothing is coming on screen (This localhost page can’t be found. No webpage was found for the web address: http://localhost:5173/
HTTP ERROR 404)
Please let me know how can the task be achieved?

**Reply by nsarrazin** [reply]
chat-ui is built as a fullstack app using SvelteKit, where the node backend handles the calls to the database (mongodb only for now) so I don't think you can easily just remove the server folder :sweat_smile: 

We support modular endpoints for inference but the app itself won't work if you remove the server. Hope that answers your question!

**Reply by parthdacoder** [reply]
Yes it answers the question. Thanks!
